tests/llmcompressor/test_kv_cache.py

```diff
@@ -0,0 +1,310 @@
+import os
+import tempfile
+
+import pytest
+from compressed_tensors.compressors import ModelCompressor
+from compressed_tensors.quantization.cache import KVCacheScaleType
+from compressed_tensors.quantization.utils.helpers import iter_named_quantizable_modules
+from datasets import load_dataset
+from transformers import AutoTokenizer
+
+from llmcompressor.transformers import SparseAutoModelForCausalLM, oneshot
+
+MODEL_IDS = [
+    "TinyLlama/TinyLlama-1.1B-Chat-v1.0",
+    "microsoft/Phi-3-mini-4k-instruct",
+]
+
+
+@pytest.fixture(scope="session")
+def oneshot_fixture():
+    num_bits = 8
+    _type = "float"
+    strategy = "tensor"
+    dynamic = False
+    symmetric = True
+
+    recipe = f"""
+    quant_stage:
+        quant_modifiers:
+            QuantizationModifier:
+                kv_cache_scheme:
+                    num_bits: {num_bits}
+                    type: {_type}
+                    strategy: {strategy}
+                    dynamic: {dynamic}
+                    symmetric: {symmetric}
+    """
+
+    used_args = dict(
+        num_bits=num_bits,
+        _type=_type,
+        strategy=strategy,
+        dynamic=dynamic,
+        symmetric=symmetric,
+    )
+
+    oneshot_args = dict(
+        dataset="open_platypus",
+        recipe=recipe,
+        num_calibration_samples=16,
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        for model_id in MODEL_IDS:
+            oneshot_args["output_dir"] = os.path.join(tmpdir, model_id)
+            used_args["output_dir"] = oneshot_args["output_dir"]
+            yield oneshot(model=model_id, **oneshot_args), used_args
+
+
+def test_kv_cache_config_format(oneshot_fixture):
+    _, used_args = oneshot_fixture
+
+    output_dir = used_args["output_dir"]
+
+    compressor = ModelCompressor.from_pretrained(output_dir)
+    # check config is properly populated
+    quant_config = compressor.quantization_config
+    assert quant_config is not None
+    assert quant_config.kv_cache_scheme is not None
+
+    # check the values match
+    scheme = quant_config.kv_cache_scheme
+    assert scheme.num_bits == used_args["num_bits"]
+    assert scheme.type == used_args["_type"]
+    assert scheme.strategy == used_args["strategy"]
+    assert scheme.dynamic == used_args["dynamic"]
+    assert scheme.symmetric == used_args["symmetric"]
+
+
+def test_kv_cache_model_state_dict_attr(oneshot_fixture):
+    model, used_args = oneshot_fixture
```

Comment:
`model` is not used. Infact, `model is None` since `oneshot` does not return anything